### ISSUE TITLE
Bug with IsTransitiveBinaryRelation?

### ISSUE THREAD

**Issue by vendramin** [gap-system/gap#4049]
### Observed behaviour

I get a funny error message that should not be there.

### Copy and paste GAP banner (to tell us about your setup)

I found something that I cannot understand. I construct a relation but I get an error while trying to check if the relation is transitive. Over the set of subsets of X={1..6} I consider the relation given by A rel B if and only if 2 does not belong to the intersection between A and the complement (in X) of B. GAP returns an error when I want to test if this relation is transitive. Here is the code:

gap> set := [1..6];;
gap> e := EnumeratorOfCombinations(set);;
gap> l := [];;
gap> for i in [1..Size(e)] do
> for j in [1..Size(e)] do
> if not 2 in Intersection(e[i], Difference(set, e[j])) then
> Add(l, Tuple([i,j]));
> fi;
> od;
> od;
gap> rel := BinaryRelationByElements(Domain(set), l);;
gap> IsTransitiveBinaryRelation(rel);
Error, <elm> must be an element of Source(<map>) at /usr/local/src/gap-4.11.0/lib/mapping.gi:243 called from
Images( m, i ) at /usr/local/src/gap-4.11.0/lib/relation.gi:285 called from
<function "IsTransitiveBinaryRelation transitive test binary relation">(\
<arguments> )
called from read-eval loop at *stdin*:15
type 'quit;' to quit to outer loop
brk>

Other GAP functions works well here:

gap> IsSymmetricBinaryRelation(rel);
false
gap> IsReflexiveBinaryRelation(rel);
true

**Comment by fingolfin** [comment]
You are defining a domain on the set `set = [1..6]`,  but the pairs you are using actually contain elements from the enumerator `e` (which contains the 64 subsets of `set`). That doesn't make sense.

So the real bug seems to be that `BinaryRelationByElements` even accepts this input.

**Comment by fingolfin** [comment]
@vendramin does that clarify things?

**Comment by vendramin** [comment]
Yes, it is. Sorry for my mistake. I found funny that others functions were returning something meaningful.
  
  
  

  
  
  

  
  
>
